test/unit/sthDatabaseNameCodecTool_test.js

```diff
@@ -275,303 +266,390 @@ function shouldEncodeOrDecodeDatabaseTest(databaseName, encodingFlag, callback)
-          true));
-
-      after(function() {
-        sthConfig.NAME_ENCODING = ORIGINAL_NAME_ENCODING;
-        sthConfig.NAME_SEPARATOR = ORIGINAL_NAME_SEPARATOR;
-      });
-      sthConfig.NAME_ENCODING = ORIGINAL_NAME_ENCODING;
-      sthConfig.NAME_SEPARATOR = ORIGINAL_NAME_SEPARATOR;
+    this.timeout(5000);
+    var ORIGINAL_NAME_ENCODING = sthConfig.NAME_ENCODING,
+        ORIGINAL_NAME_SEPARATOR = sthConfig.NAME_SEPARATOR;
+
+    describe('getEncodingAnalysis tests', function() {
+        describe('name encoding disabled', function() {
+            sthConfig.NAME_ENCODING = false;
+            sthConfig.NAME_SEPARATOR = '_';
+            before(function(done) {
+                sthConfig.NAME_ENCODING = false;
+                sthConfig.NAME_SEPARATOR = '_';
+                async.series([connectToDatabase, dropDatabases, createRawAndAggregatedCollections], done);
+            });
+
+            it(
+                'should raise an error if neither -e, --encode or -d, --decode options are set',
+                encodeDecodeMandatoryOptionsTest.bind(null, sthDatabaseNameCodecTool.getEncodingAnalysis)
+            );
+
+            it(
+                'should raise an error if the -b, --database is not set when the -c, --collection is set',
+                collectionAndDatabaseMandatoryOptionsTest.bind(null, sthDatabaseNameCodecTool.getEncodingAnalysis)
+            );
+
+            it("should detect database '" + DATABASE_NAME + "' as susceptible of being encoded", function(done) {
+                sthDatabaseNameCodecTool.getEncodingAnalysis(
+                    {
+                        encode: true
+                    },
+                    function(err, analysis) {
+                        if (err) {
+                            return done(err);
+                        }
+                        expect(analysis[DATABASE_NAME].name).to.not.equal(DATABASE_NAME);
+                        done();
+                    }
+                );
+            });
+
+            it(
+                "should not detect database '" + DATABASE_NAME + "' as susceptible of being encoded if filtered out",
+                function(done) {
+                    sthDatabaseNameCodecTool.getEncodingAnalysis(
+                        {
+                            encode: true,
+                            database: sthConfig.DB_PREFIX + 'inexistent_database'
+                        },
+                        function(err, analysis) {
+                            if (err) {
+                                return done(err);
+                            }
+                            expect(analysis[DATABASE_NAME]).to.be(undefined);
+                            done();
+                        }
+                    );
+                }
+            );
+
+            it(
+                "should detect collection '" +
+                    sthDatabaseNaming.getRawCollectionName(COLLECTION_NAME_PARAMS) +
+                    "' of database '" +
```

Comment:
Excesive line break here and in other places in this file. Please do a general review.

Reply:
Fixed  c00464eb